File: core-java/src/main/java/com/kakao/actionbase/core/java/metadata/v3/common/Direction.java
package com.kakao.actionbase.core.java.metadata.v3.common;

public enum Direction {
  // A --> B
  OUT((byte) 2),

  // A <-- B
  IN((byte) 3);

  private final byte code;

  Direction(byte code) {
    this.code = code;
  }

  public byte getCode() {
    return code;
  }

  public static Direction fromCode(byte code) {
    for (Direction direction : values()) {
      if (direction.code == code) {
        return direction;
      }
    }
    throw new IllegalArgumentException("Invalid code: " + code);
  }
}

File: core/src/main/kotlin/com/kakao/actionbase/core/metadata/common/Direction.kt
package com.kakao.actionbase.core.metadata.common

enum class Direction(
    val code: Byte,
) {
    OUT(2),
    IN(3),
    ;

    companion object {
        private val DIRECTION_BY_CODE = entries.associateBy(Direction::code)

        fun of(code: Byte): Direction = DIRECTION_BY_CODE[code] ?: throw IllegalArgumentException("Invalid direction code: $code")
    }
}
